src/main/java/domain/WinningStatistics.java

```diff
@@ -0,0 +1,52 @@
+package domain;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+
+import constant.LottoConstant;
+
+public class WinningStatistics {
+
+    private static final int DEFAULT_VALUE = 0;
+
+    private final Map<LottoReward, Integer> statistics = new EnumMap<>(LottoReward.class);
+
+    public WinningStatistics(List<LottoReward> lottoRewards) {
+        validateNull(lottoRewards);
+        Arrays.stream(LottoReward.values()).forEach(lottoReward -> statistics.put(lottoReward, DEFAULT_VALUE));
+        lottoRewards.forEach(lottoReward -> statistics.replace(lottoReward, statistics.get(lottoReward) + 1));
+    }
+
+    private void validateNull(List<LottoReward> lottoRewards) {
+        if (lottoRewards == null) {
+            throw new NullPointerException("null 로 WinningStatistics 를 생성할 수 없습니다.");
+        }
+    }
+
+    public double calculateProfitRate() {
+        int winningAmount = calculateWinningAmount();
+        int purchasedLottoAmount = calculatePurchasedLottoAmount();
+
+        return (double)winningAmount / purchasedLottoAmount;
+    }
+
+    private int calculateWinningAmount() {
+        return statistics.entrySet().stream()
+            .map(entry -> entry.getKey().getPrice() * entry.getValue())
```

Comment:
혹시 해당 리워드의 당첨금액을  `getPrice()`로 직접 계산하는 것이 아니라 getter를 숨기고 `lottoReward.calculate(rewardCount)`와 같이 추가해서 lottoReward 가 직접 계산하도록 하는 것이 더 좋은 방법일까요?
아니면 rewardCount 는 리워드가 가지고 있을 정보는 아니기 때문에 외부에서 판단해도 무방하려나요? 🤔

Reply:
음 제 생각에는 LottoReward는 당첨 금액과 관련된 정보를 제공하고 계산은 계산을 담당하는 Statistics에서 해주는게 역할면에서 더 나을 것 같아요

getter를 두번 사용하는게 걸린다면 이런식으로 계산할수도 있을 것 같아요~

```suggestion
       	return Arrays.stream(LottoReward.values())
			.mapToInt(reward -> reward.getPrice() * statistics.get(reward))
			.sum();
```

Reply:
오 넵넵 좋은 예시 알려주셔서 감사합니다 !! 😄